backend/src/main/java/fk/prof/backend/aggregator/AggregationWindow.java

```diff
@@ -69,27 +70,37 @@ public AggregationState abandonProfile(long workId) throws AggregationFailure {
   }
 
   /**
-   * Aborts all in-flight profiles. Should be called when aggregation window expires
+   * Called by backend daemon thread
+   * Does following tasks required to expire aggregation window:
+   * > Marks status of ongoing profiles as aborted
+   * > De associates assigned work with this aggregation window
+   * > Updates ended at time for aggregation window
+   * > Finalizes the window
+   * @param aggregationWindowLookupStore
+   * @return finalized aggregation window
    */
-  public void abortOngoingProfiles() throws AggregationFailure {
+  public FinalizedAggregationWindow expireWindow(AggregationWindowLookupStore aggregationWindowLookupStore) {
     ensureEntityIsWriteable();
 
-    for (Map.Entry<Long, ProfileWorkInfo> entry : workInfoLookup.entrySet()) {
-      try {
-        entry.getValue().abortProfile();
-      } catch (IllegalStateException ex) {
-        throw new AggregationFailure(String.format("Error aborting profile for work_id=%d", entry.getKey()), ex);
-      }
-    }
+    abortOngoingProfiles();
+    long[] workIds = this.workInfoLookup.keySet().stream().mapToLong(Long::longValue).toArray();
+    aggregationWindowLookupStore.deAssociateAggregationWindow(workIds);
+    this.endedAt = LocalDateTime.now(Clock.systemUTC());
```

Comment:
start is assigned in the constructor itself (maybe I can not accept that as a param and generate value in constructor itself). Unless clock goes backward, end first and start later cannot happen, right.